include/zephyr/bluetooth/conn.h

```diff
@@ -1849,12 +1856,16 @@ struct bt_conn_cb {
 	 * This callback notifies the application that it might be able to
 	 * allocate a connection object. No guarantee, first come, first serve.
 	 *
-	 * Use this to e.g. re-start connectable advertising or scanning.
+	 * The maximum number of simultaneous connections is configured
+	 * by @kconfig{CONFIG_BT_MAX_CONN}, which usually defaults to a single connection.
+	 *
+	 * This is the event to listen for to start a new connection, connectable advertiser
+	 * or scanner, both when the intention is to start it after the system is completely
+	 * finished with an earlier connection, and when the application wants to start
+	 * a connection for any reason but failed and is waiting for the right time to retry.
 	 *
-	 * Treat this callback as an ISR, as it originates from
-	 * @ref bt_conn_unref which is used by the BT stack. Making
-	 * Bluetooth API calls in this context is error-prone and strongly
-	 * discouraged.
+	 * This callback runs on the system workqueue thread,
+	 * the usual precautions apply.
```

Comment:
What are the usual precautions? I think that if we want to write such as statement, we should define those and refer to it. If you don't know the usual precautions, then this isn't helpful :)